Source/Converters/fsEnumConverter.cs

```diff
@@ -26,26 +26,31 @@ public class fsEnumConverter : fsConverter {
         }
 
         public override fsResult TrySerialize(object instance, out fsData serialized, Type storageType) {
-            if (fsPortableReflection.GetAttribute<FlagsAttribute>(storageType) != null) {
-                long instanceValue = Convert.ToInt64(instance);
-                var result = new StringBuilder();
-
-                bool first = true;
-                foreach (var value in Enum.GetValues(storageType)) {
-                    int integralValue = (int)value;
-                    bool isSet = (instanceValue & integralValue) != 0;
-
-                    if (isSet) {
-                        if (first == false) result.Append(",");
-                        first = false;
-                        result.Append(value.ToString());
-                    }
-                }
-
-                serialized = new fsData(result.ToString());
+            if (fsConfig.SerializeEnumsAsInteger) {
+                serialized = new fsData(Convert.ToInt64(instance));
             }
             else {
-                serialized = new fsData(Enum.GetName(storageType, instance));
```

Comment:
Can this entire block be shifted into an `else if` instead of a nested `else`?

So the structure will look like:

```
if (blah) { ... }
else if (blah) { ... }
else { ... }
```